volttrontesting/platform/dbutils/test_backup_database.py

```diff
@@ -57,8 +59,69 @@ def test_backup_new_data_should_filter_duplicates(backup_database):
     assert actual_results == expected_results
 
 
+def test_remove_successfully_published_should_delete_by_ts_topic_id(backup_database):
+    new_publish_list = [
+        {
+            "source": "dupesource",
+            "topic": "dupetopic",
+            "meta": {},
+            "readings": [("2020-06-01 12:30:59", 123)],
+            "headers": {},
+        },
+        {
+            "source": "dupesource",
+            "topic": "dupetopic",
+            "meta": {},
+            "readings": [("2020-06-01 12:30:59", 123)],
+            "headers": {},
+        },
+        {
+            "source": "foobar_source",
+            "topic": "foobar_topic",
+            "meta": {},
+            "readings": [("2020-05-25 08:46:00", 846)],
+            "headers": {},
+        },
+    ]
+    backup_database.backup_new_data(new_publish_list)
+    outstanding_before = [
+        "1|2020-06-01 12:30:59|dupesource|1|123|{}",
+        "2|2020-06-01 12:30:59|dupesource|1|123|{}",
+        "3|2020-05-25 08:46:00|foobar_source|2|846|{}",
+    ]
+    assert outstanding_before == get_all_data("outstanding")
+
+    successful_publishes = {("2020-06-01 12:30:59", "dupetopic")}
+    expected_outstanding_after = ["3|2020-05-25 08:46:00|foobar_source|2|846|{}"]
+
+    backup_database.remove_successfully_published(successful_publishes, 1000)
+
+    assert expected_outstanding_after == get_all_data("outstanding")
```

Comment:
I am wondering use id(expected_outstanding_after) != id(get_all_data("outstanding")) is true...the references in list could be the same and it shouldn't be after the serialize/deserialize from the database.

Reply:
I just tried checking the identity of both both object and yes, `id(expected_outstanding_after) != id(get_all_data("outstanding")) is true as expected. Then I checked the id's inside both lists and they are not the same either. Are you suggesting to check for reference on top of value in order to ensure that both objects are not the same thing?